x-pack/plugins/security/server/config.ts

```diff
@@ -56,6 +53,23 @@ export const ConfigSchema = schema.object(
   { allowUnknowns: true }
 );
 
+// HACK: Since new platform doesn't support config deprecation transformations yet (e.g. `rename`), we have to handle
+// them manually here for the time being. Legacy platform config will log corresponding deprecation warnings.
+const origValidate = ConfigSchema.validate;
```

Comment:
__note:__ super hacky, but feels like the safest and correct for the time being.